studio/data/organizations/organization-tax-ids-update-mutation.ts

```diff
@@ -0,0 +1,82 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { delete_, post } from 'lib/common/fetch'
+import { API_URL } from 'lib/constants'
+import { organizationKeys } from './keys'
+import { TaxId } from './organization-tax-ids-query'
+
+export interface TaxIdValue {
+  id: string
+  type: string
+  value: string
+  name: string
+}
+
+export type OrganizationTaxIDsUpdateVariables = {
+  slug: string
+  existingIds: TaxId[]
+  newIds: TaxIdValue[]
+}
+
+export async function updateOrganizationTaxIDs({
+  slug,
+  existingIds,
+  newIds,
+}: OrganizationTaxIDsUpdateVariables) {
+  if (!slug) {
+    console.error('Slug is required')
+    return {}
+  }
+
+  // To make things simple we delete all existing ones and create new ones from this session
+  if (existingIds.length > 0) {
+    await Promise.all(
+      existingIds.map(async (taxId) => {
+        return await delete_(`${API_URL}/organizations/${slug}/tax-ids`, { id: taxId.id })
```

Comment:
This `await` doesn't block on each loop, right (because of the `async`)? Just want to make sure 😬

Reply:
yup! the deletes will run in parallel